controllers/apps/apicast_controller.go

```diff
@@ -142,8 +140,13 @@ func (r *APIcastReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&appsv1alpha1.APIcast{}).
 		Watches(
-			&source.Kind{Type: &v1.Secret{}},
-			handler.EnqueueRequestsFromMapFunc(secretToApicastEventMapper.Map),
+			&corev1.Secret{},
+			handler.EnqueueRequestsFromMapFunc(
+				func(ctx context.Context, obj client.Object) []reconcile.Request {
+					secret := &corev1.Secret{}
+					reqs := secretToApicastEventMapper.Map(secret)
```

Comment:
this does not look good. Unless I am missing something, the `secretToApicastEventMapper.Map` function needs the `obj` object. Mainly because it needs the UID and that secret is empty.

Reply:
The 3scale operator implementation seems correct in this regard. 

You can try. The operator will add some labels with the UID of the secret to the APIcast CR. . Try updating the secret and see if that triggers a new reconciliation event for APIcast CR.